x-pack/plugins/apm/server/routes/metrics/by_agent/serverless/compute_usage.ts

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { i18n } from '@kbn/i18n';
+import { ProcessorEvent } from '@kbn/observability-plugin/common';
+import { euiLightVars as theme } from '@kbn/ui-theme';
+import {
+  kqlQuery,
+  rangeQuery,
+  termQuery,
+} from '@kbn/observability-plugin/server';
+import {
+  FAAS_BILLED_DURATION,
+  FAAS_COLDSTART_DURATION,
+  FAAS_ID,
+  METRIC_SYSTEM_TOTAL_MEMORY,
+  SERVICE_NAME,
+} from '../../../../../common/elasticsearch_fieldnames';
+import { environmentQuery } from '../../../../../common/utils/environment_query';
+import { getMetricsDateHistogramParams } from '../../../../lib/helpers/metrics';
+import { Setup } from '../../../../lib/helpers/setup_request';
+import { ChartBase } from '../../types';
+import { getVizColorForIndex } from '../../../../../common/viz_colors';
+import { GenericMetricsChart } from '../../fetch_and_transform_metrics';
+
+const chartBase: ChartBase = {
+  title: i18n.translate('xpack.apm.agentMetrics.serverless.computeUsage', {
+    defaultMessage: 'Compute usage',
+  }),
+  key: 'compute_usage',
+  type: 'linemark',
+  yUnit: 'number',
+  series: {
+    computeUsage: {
+      title: i18n.translate('xpack.apm.agentMetrics.serverless.computeUsage', {
+        defaultMessage: 'Compute usage',
+      }),
+    },
+  },
+};
+
+/**
+ * To calculate the compute usage we need to multiple the "system.memory.total" by "faas.billed_duration".
+ * But the result of this calculation is in Bytes-milliseconds, as the "system.memory.total" is stored in bytes and the "faas.billed_duration" is stored in milliseconds.
+ * But to calculate the overall cost AWS uses GB-second, so we need to convert the result to this unit.
+ */
+function calculateComputeUsageGBSeconds({
+  faasBilledDuration,
+  totalMemory,
+}: {
+  faasBilledDuration?: number | null;
+  totalMemory?: number | null;
+}) {
+  if (
+    faasBilledDuration === undefined ||
+    faasBilledDuration === null ||
+    totalMemory === undefined ||
+    totalMemory === null
+  ) {
+    return 0;
+  }
```

Comment:
you can also use `isFiniteNumber` to refine. saves you a line or two. doesn't really matter though I think.

Reply:
I'll have to check for undefined or null anyway because of TS.

Reply:
No need, isFiniteNumber also refines to the number type for typescript. If not, it's a bug.

Reply:
yeah, you're right. Thanks.